echo_server.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+import socket
+
+
+server_socket = socket.socket(
+    socket.AF_INET,
+    socket.SOCK_STREAM,
+    socket.IPPROTO_IP)
+
+buffer_size = 32
+
+server_socket.bind(('127.0.0.1', 50000))
+server_socket.listen(1)
+
+
+try:
+    while True:
+        msg = 'Echo: "'
+        conn, addr = server_socket.accept()
+
+        msg_part = conn.recv(buffer_size).decode('utf-8')
+        msg += msg_part
+        while len(msg_part) >= buffer_size:
+            msg_part = conn.recv(buffer_size).decode('utf-8')
+            msg += msg_part
```

Comment:
Lines 21 & 22 and lines 24 & 25 are identical.  This is because you are using the wrong kind of test for terminating your while loop.  Can you fix this?